src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SamlSecurityTokenHandler.cs

```diff
@@ -0,0 +1,200 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Security.Claims;
+using System.Xml;
+using CoreWCF.IdentityModel.Selectors;
+using Microsoft.IdentityModel.Tokens.Saml;
+using MSIdentityTokens = Microsoft.IdentityModel.Tokens;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    /// <summary>
+    /// This class implements a SecurityTokenHandler for a Saml11 token.  It contains functionality for: Creating, Serializing and Validating 
+    /// a Saml 11 Token.
+    /// </summary>
+    public class SamlSecurityTokenHandler : SecurityTokenHandler
+    {
+        private static readonly string[] s_tokenTypeIdentifiers = new string[] { SecurityTokenTypes.SamlTokenProfile11, SecurityTokenTypes.OasisWssSamlTokenProfile11 };
+        SecurityTokenSerializer _keyInfoSerializer;
+        private readonly MSIdentityTokens.Saml.SamlSecurityTokenHandler _internalSamlSecurityTokenHandler;
+        private readonly object _syncObject = new object();
+
+        /// <summary>
+        /// Initializes an instance of <see cref="SamlSecurityTokenHandler"/>
+        /// </summary>
+        public SamlSecurityTokenHandler()
+            //: this(new SamlSecurityTokenRequirement())
+        {
+            _internalSamlSecurityTokenHandler = new MSIdentityTokens.Saml.SamlSecurityTokenHandler();
+        }
+
+        #region TokenValidation
+        /// <summary>
+        /// Returns value indicates if this handler can validate tokens of type
+        /// SamlSecurityToken.
+        /// </summary>
+        public override bool CanValidateToken
+        {
+            get { return _internalSamlSecurityTokenHandler.CanValidateToken; }
+        }
+
+        /// <summary>
+        /// Validates a <see cref="SamlSecurityToken"/>.
+        /// </summary>
+        /// <param name="token">The <see cref="SamlSecurityToken"/> to validate.</param>
+        /// <returns>The <see cref="ReadOnlyCollection{T}"/> of <see cref="ClaimsIdentity"/> representing the identities contained in the token.</returns>
+        /// <exception cref="ArgumentNullException">The parameter 'token' is null.</exception>
+        /// <exception cref="ArgumentException">The token is not assignable from <see cref="SamlSecurityToken"/>.</exception>
+        /// <exception cref="InvalidOperationException">Configuration <see cref="SecurityTokenHandlerConfiguration"/>is null.</exception>
+        /// <exception cref="ArgumentException">SamlSecurityToken.Assertion is null.</exception>
+        /// <exception cref="SecurityTokenValidationException">Thrown if SamlSecurityToken.Assertion.SigningToken is null.</exception>
+        /// <exception cref="SecurityTokenValidationException">Thrown if the certificate associated with the token issuer does not pass validation.</exception>
+        public override ReadOnlyCollection<ClaimsIdentity> ValidateToken(SecurityToken token)
+        {
```

Comment:
The changes for Saml2SecurityTokenHandler also need to happen in this class where appropriate